lib/fpm/package/cpan.rb

```diff
@@ -150,6 +150,10 @@ def input(package)
        found_dependencies.each do |dep_name, version|
           # Special case for representing perl core as a version.
           if dep_name == "perl"
+            m = version.match(/^(\d)\.(\d{3})(\d{3})$/)
+            if m
+               version = m[1] + '.' + m[2].sub(/^0*/, '') + '.' + m[3].sub(/^0*/, '')
```

Comment:
Well, I have a bit of ignorance here, but my Ubuntu machine has Perl `5.22.2` installed, not 5.022002`, and I would presume that a comparison of similar x.y.z versions would be better there, as well.